python/spacetime/store.py

```diff
@@ -19,20 +19,36 @@
 import uuid
 import logging
 
+def calc_basecls2derived(name2class):
+    result = {}
+    for tp in DATAMODEL_TYPES:
+        if hasattr(tp, "__PCC_BASE_TYPE__") and tp.__PCC_BASE_TYPE__:
+            for base in tp.mro():
+                if (base != tp.Class()
+                        and base.__name__ in name2class 
+                        and hasattr(name2class[base.__name__], "__PCC_BASE_TYPE__") 
+                        and name2class[base.__name__].__PCC_BASE_TYPE__):
+                    result.setdefault(name2class[base.__name__], []).append(tp)
+    #                print base.__name__, tp.__name__
+    #print [(tp.Class().__name__, [der.Class().__name__ for der in result[tp]]) for tp in result]
+    return result
+
+
 
 # not active object.
 # just stores the basic sets.
 # requires set types that it has to store.
 # must accept changes.
 # must return object when asked for it.
 class store(object):
-    def __init__(self):
+    def __init__(self, basecls2derived):
```

Comment:
basecls2derived shouldn't be a parameter of init, but calculated inside using DATAMODEL_TYPES